Mixpanel/MPNetwork.m

```diff
@@ -18,6 +18,18 @@
 
 @implementation MPNetwork
 
++ (NSURLSession *)sharedSession {
+    static NSURLSession *sharedSession = nil;
+    @synchronized(self) {
+        if (sharedSession == nil) {
+            NSURLSessionConfiguration *sessionConfig = [NSURLSessionConfiguration defaultSessionConfiguration];
+            sessionConfig.timeoutIntervalForRequest = 5.0;
```

Comment:
defaultSessionConfiguration is a factory method that creates a copy of the default configuration and doesn't alter it